packable/packable-derive/src/fragments.rs

```diff
@@ -32,21 +32,39 @@ impl Fragments {
             fields_type,
         } = info;
 
-        let fields_verification = fields_verify_with.into_iter().zip(fields_ident.iter()).map(|(verify_with, field_ident)| match verify_with {
-            Some(verify_with) => if unpack_visitor_info.explicit {
-                quote!(#verify_with::<VERIFY>(&#field_ident, visitor).map_err(#crate_name::error::UnpackError::from_packable)?;)
-            } else {
-                quote!(#verify_with::<VERIFY>(&#field_ident).map_err(#crate_name::error::UnpackError::from_packable)?;)
-            }
-            None => quote!(),
-        });
+        let fields_verification = fields_verify_with.into_iter().zip(fields_ident.iter()).map(
+            |(verify_with, field_ident)| match verify_with {
+                Some(verify_with) => if unpack_visitor_info.explicit {
+                    quote! {
+                        if let Some(visitor) = visitor {
+                            #verify_with(&#field_ident, visitor).map_err(#crate_name::error::UnpackError::from_packable)?;
+                        }
+                    }
+                } else {
+                    quote! {
+                        if visitor.is_some() {
+                            #verify_with(&#field_ident).map_err(#crate_name::error::UnpackError::from_packable)?;
+                        }
```

Comment:
Can you maybe add some comment in the code why this is done like this?

Reply:
I mean...it's in the derive crate so I don't know if there's any point.